docs/sdks/typescript/index.md

```diff
@@ -55,927 +67,740 @@ Each SpacetimeDB client depends on some bindings specific to your module. Create
+| [`onConnect` callback](#callback-onconnect)           | Register a callback to run when the connection is successfully established.          |
+| [`onConnectError` callback](#callback-onconnecterror) | Register a callback to run if the connection is rejected or the host is unreachable. |
+| [`onDisconnect` callback](#callback-ondisconnect)     | Register a callback to run when the connection ends.                                 |
+| [`withToken` method](#method-withtoken)               | Supply a token to authenticate with the remote database.                             |
+| [`build` method](#method-build)                       | Finalize configuration and connect.                                                  |
 
-```ts
-const host = 'ws://localhost:3000';
-const name_or_address = 'database_name';
-const auth_token = undefined;
-const protocol = 'binary';
+#### Method `withUri`
 
-var spacetimeDBClient = new SpacetimeDBClient(
-  host,
-  name_or_address,
-  auth_token,
-  protocol
-);
+```typescript
+DbConnectionBuilder.withUri(uri: string): DbConnectionBuilder
 ```
 
-## Class methods
+Configure the URI of the SpacetimeDB instance or cluster which hosts the remote module.
 
-### `SpacetimeDBClient.registerReducers`
+#### Method `withModuleName`
 
-Registers reducer classes for use with a SpacetimeDBClient
+```typescript
+DbConnectionBuilder.withModuleName(name_or_identity: string): DbConnectionBuilder
 
-```ts
-registerReducers(...reducerClasses: ReducerClass[])
 ```
 
-#### Parameters
+Configure the SpacetimeDB domain name or `Identity` of the remote module which identifies it within the SpacetimeDB instance or cluster.
 
-| Name             | Type           | Description                   |
-| :--------------- | :------------- | :---------------------------- |
-| `reducerClasses` | `ReducerClass` | A list of classes to register |
+#### Callback `onConnect`
 
-#### Example
-
-```ts
-import SayHelloReducer from './types/say_hello_reducer';
-import AddReducer from './types/add_reducer';
-
-SpacetimeDBClient.registerReducers(SayHelloReducer, AddReducer);
+```typescript
+DbConnectionBuilder.onConnect(
+    callback: (ctx: DbConnection, identity: Identity, token: string) => void
+): DbConnectionBuilder
 ```
 
----
+Chain a call to `.onConnect(callback)` to your builder to register a callback to run when your new `DbConnection` successfully initiates its connection to the remote module. The callback accepts three arguments: a reference to the `DbConnection`, the `Identity` by which SpacetimeDB identifies this connection, and a private access token which can be saved and later passed to [`withToken`](#method-withtoken) to authenticate the same user in future connections.
 
-### `SpacetimeDBClient.registerTables`
+This interface may change in an upcoming release as we rework SpacetimeDB's authentication model.
```

Comment:
Will it? I was not aware of planned changes. But also I don't think we should say this, because this either saying it's not stable (even though we're claiming 1.0) or it's coming in 2.0 in which case we should just not say it.

Reply:
Ah okay, that makes more sense.